applications/Preprocessor/lib/mesh/MeshImpl/Mesh_Impl.h

```diff
@@ -234,8 +229,62 @@ void Mesh<dimension>::fixEntity(Element<dimension>& element,
     EntityGId entityIndex =
         (candidates.size() == 1 ? candidates[0] : newEntity<d>());
     element.template addEntity<d>(entityIndex);
-    meshEntities.template get<d>()[entityIndex.id].addElement(
-        element.getGlobalIndex(), index);
+}
+
+template <std::size_t dimension>
+template <int d>
+void Mesh<dimension>::removeUnusedEntities(itag<d> dimTag) {
+    std::vector<EntityGId> renumbering;
+    // Fill renumbering with invalid data
+    std::size_t currentCount = this->meshEntities[dimTag].size();
+    renumbering.resize(currentCount, EntityGId(-1));
```

Comment:
Can we do it in the constructor of the std::vector?

Reply:
Yes